clippy_lints/src/manual_is_power_of_two.rs

```diff
@@ -0,0 +1,88 @@
+use clippy_utils::diagnostics::span_lint_and_sugg;
+use clippy_utils::source::snippet_with_applicability;
+use rustc_ast::LitKind;
+use rustc_data_structures::packed::Pu128;
+use rustc_errors::Applicability;
+use rustc_hir::{BinOpKind, Expr, ExprKind};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_middle::ty::Uint;
+use rustc_session::declare_lint_pass;
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for expressions like `x.count_ones() == 1` or `x & (x - 1) == 0` which are manual
+    /// reimplementations of `x.is_power_of_two()``
```

Comment:
Minor punctuation changes:
```suggestion
    /// Checks for expressions like `x.count_ones() == 1` or `x & (x - 1) == 0`, which are manual
    /// reimplementations of `x.is_power_of_two()`.
```